test/org/sagebionetworks/bridge/services/ParticipantServiceTest.java

```diff
@@ -320,26 +327,33 @@ public void createParticipant() {
         STUDY.setEmailVerificationEnabled(true);
         mockHealthCodeAndAccountRetrieval();
         
-        when(externalIdService.getExternalId(STUDY.getStudyIdentifier(), EXTERNAL_ID)).thenReturn(Optional.of(EXT_ID));
+        when(externalIdService.getExternalId(STUDY.getStudyIdentifier(), EXTERNAL_ID)).thenReturn(Optional.of(extId));
         
         IdentifierHolder idHolder = participantService.createParticipant(STUDY, PARTICIPANT, true);
         assertEquals(ID, idHolder.getIdentifier());
         
-        verify(accountDao).constructAccount(STUDY, EMAIL, PHONE, EXTERNAL_ID, PASSWORD);
-        verify(externalIdService).commitAssignExternalId(EXT_ID);
+        verify(externalIdService).commitAssignExternalId(extId);
         
         // suppress email (true) == sendEmail (false)
         verify(accountDao).createAccount(eq(STUDY), accountCaptor.capture(), any());
         verify(accountWorkflowService).sendEmailVerificationToken(STUDY, ID, EMAIL);
         
         Account account = accountCaptor.getValue();
+        assertEquals(STUDY.getIdentifier(), account.getStudyId());
+        assertEquals(EMAIL, account.getEmail());
+        assertEquals(PHONE, account.getPhone());
+        assertEquals(EXTERNAL_ID, account.getExternalId());
+        assertNotNull(account.getPasswordHash());
+        assertEquals(PasswordAlgorithm.DEFAULT_PASSWORD_ALGORITHM, account.getPasswordAlgorithm());
+        assertNotEquals(PASSWORD, account.getPasswordHash());
```

Comment:
We should also check for the following attributes: id, phoneVerified, healthCode.

Reply:
Added